cheetah/accelerator/transverse_deflecting_cavity.py

```diff
@@ -55,23 +58,23 @@ def __init__(
             (
                 torch.as_tensor(voltage, **factory_kwargs)
                 if voltage is not None
-                else torch.tensor(0.0, **factory_kwargs)
+                else torch.zeros_like(self.length)
             ),
         )
         self.register_buffer(
             "phase",
             (
                 torch.as_tensor(phase, **factory_kwargs)
                 if phase is not None
-                else torch.tensor(0.0, **factory_kwargs)
+                else torch.zeros_like(self.length)
             ),
         )
         self.register_buffer(
             "frequency",
             (
                 torch.as_tensor(frequency, **factory_kwargs)
                 if frequency is not None
-                else torch.tensor(0.0, **factory_kwargs)
+                else torch.zeros_like(self.length)
             ),
         )
         self.register_buffer(
```

Comment:
I think these lines are causing / exposing the issue. They should definitely be like
```python
torch.tensor(0.0, **factory_kwargs)
```
And then we should check if vectorising things like `frequency` works, because vectorising them by mistake seems to have caused the test failures.

Reply:
I agree. This will likely fix the tests for this PR, but its effectively just hiding the issue.